service/frontend.js

```diff
@@ -127,16 +127,16 @@ module.exports = class WebFrontend extends events.EventEmitter {
         });
 
         // mount /api before csurf so we can perform requests without the CSRF token
-        this._app.use('/api', require('../routes/api'));
+        this._app.use(Config.BASE_URL + '/api', require('../routes/api'));
 
         this._app.use(csurf({ cookie: false,
                               ignoreMethods: ['GET','HEAD','OPTIONS',
                                               'UPGRADE','CONNECT'] }));
-        this._app.use('/', require('../routes/index'));
-        this._app.use('/apps', require('../routes/apps'));
-        this._app.use('/user', require('../routes/user'));
-        this._app.use('/config', require('../routes/config'));
-        this._app.use('/devices', require('../routes/devices'));
+        this._app.use(Config.BASE_URL + '/', require('../routes/index'));
+        this._app.use(Config.BASE_URL + '/apps', require('../routes/apps'));
+        this._app.use(Config.BASE_URL + '/user', require('../routes/user'));
+        this._app.use(Config.BASE_URL + '/config', require('../routes/config'));
+        this._app.use(Config.BASE_URL + '/devices', require('../routes/devices'));
```

Comment:
The server should listen to `/api` or `/..`. Only the path for frontend need use the base path or the redirect

Reply:
Thanks for taking a look. Yeah I'm not opposed to either option. If it's more convenient for you to have Almond expose the API always at /api I'll do that.

Reply:
I mean all changes look fine until they from this file.

Reply:
Application routing is still the same behind a proxy. Only redirects or html templates need to be relative or use base path for working.

Reply:
Yeah but that makes the patch untestable, unless one sets up a proxy.
Can you configure Home Assistant to keep the path when proxy, ie proxy /almond to 127.0.0.1:3000/almond ?

Reply:
The issue is, the request comes outside with i.e. `/api/ingress/234kj234j23l/config` but your server is called with `/config`. That can solve with use relative URLs on frontend but you use a lot of redirects and this did not work with relative URLs.

To run behind a proxy mean, that the application runs like on all other installation, but the links on html/redirects are using a base url or (did not work with redirects) relative path on the frontend.

Reply:
> Yeah but that makes the patch untestable, unless one sets up a proxy.

I don't think so, because the default base path will be `/` and you only modify it if you run behind a proxy. To run application behind a proxy is a very common on docker/kubernetes.

Reply:
However, I can take this branch as a custom patch and modify our installation. However, because you provide a docker, the requests to support reverse proxy systems for Kubernetes or docker comes anyway.

Reply:
And last :) We can use the Nginx on an add-on to rewrite the URL back to URL outside. That will work with this PR and I'm als fine to do this. However, that means this patch is only useful for us. If you think that is fine, we can merge this PR like it is 👍

Reply:
> > Yeah but that makes the patch untestable, unless one sets up a proxy.
> 
> I don't think so, because the default base path will be `/` (or in your case '') and you only modify it if you run behind a proxy. To run application behind a proxy is a very common on docker/kubernetes.

Yeah, I see what you mean (in kubernetes the rewrite-target is apparently non-standard and controller-specific...). I'll guess we'll live with this not being testable if BASE_URL is set.